tests/unit/opensearch_utils/test_client.py

```diff
@@ -183,6 +197,79 @@ def test_send_payload(self):
         self.client.send_document(document1, Action.DELETE)
         self.client.send_document(document2, Action.DELETE)
 
+    def test_search(self):
+        """
+        test that the search method correctly queries the OpenSearch cluster and received the 
+        intended results.
+        """
+        ## Arrange ##
+        search_true = [{'_index': 'test_data', '_type': '_doc', '_id': '1', '_score': 0.5753642, '_source': {'mission': 'imap', 'level': 'l0', 'instrument': 'mag', 'date': '20230112', 'version': '*', 'extension': 'fits'}}]
+        self.payload.add_documents([self.document1, self.document2, self.document3])
+        self.client.send_payload(self.payload)
+        query = Query({"level":"l0", "instrument":"mag", "start_date":"20230101", "end_date":"20230201"})
+        # need to give opensearch a second to receive the payload before searching
+        time.sleep(1)
+        ## Act ##
+        search_out = self.client.search(query, self.index)
+
+        ## Assert ##
+        assert search_out == search_true
+
+        ## Teardown ##
+        self.client.send_document(self.document1, action_override=Action.DELETE)
+        self.client.send_document(self.document2, action_override=Action.DELETE)
+        self.client.send_document(self.document3, action_override=Action.DELETE)
+
+    def test_scroll_search(self):
+        """
+        test that the search method correctly scrolls through a large set of results
+        and returns all results without error
+        """
+        ## Arrange ##
+        search_true = [{'_index': 'test_data', '_type': '_doc', '_id': '3', '_score': 1.0, '_source': {'test body': 10}}, 
```

Comment:
Are they always in this order? It seems like the indexes jump around.

Reply:
The keys have always been in the same order since I've been running tests